worker/worker.go

```diff
@@ -173,6 +173,12 @@ type (
 		Run() error
 	}
 
+	AuthorizationProvider interface {
```

Comment:
This is giving me a cyclic error @longquanzheng.

If I would like to place this on `"go.uber.org/cadence/.gen/go/shared"` but I'm wondering if everything inside of `.gen` is generated and if that's the case how could I generate that struct there? Or where do you recommend me to place it?

Reply:
Yeah don’t put it under .gen. There should be other places to put it . Would it work if you put it under auth/ ? And then use type aliases to export to public